docs/widgets/text_area.md

```diff
@@ -0,0 +1,468 @@
+
+# TextArea
+
+!!! tip "Added in version 0.38.0"
+
+A widget for editing text which may span multiple lines.
+Supports syntax highlighting for a selection of languages.
+
+- [x] Focusable
+- [ ] Container
+
+
+## Guide
+
+### Loading text
+
+In this example we load some initial text into the `TextArea`, and set the language to `"python"` to enable syntax highlighting.
+
+=== "Output"
+
+    ```{.textual path="docs/examples/widgets/text_area_example.py" columns="42" lines="8"}
+    ```
+
+=== "text_area_example.py"
+
+    ```python
+    --8<-- "docs/examples/widgets/text_area_example.py"
+    ```
+
+To load content into the `TextArea` after it has already been created,
+use the [`load_text`][textual.widgets._text_area.TextArea.load_text] method.
+
+To update the parser used for syntax highlighting, set the [`language`][textual.widgets._text_area.TextArea.language] reactive attribute:
+
+```python
+# Set the language to Markdown
+text_area.language = "markdown"
+```
+
+!!! note
+    Syntax highlighting is unavailable on Apple Silicon machines running Python 3.7.
+
+!!! note
+    More built-in languages will be added in the future. For now, you can [add your own](#adding-support-for-custom-languages).
+
+
+### Reading content from `TextArea`
+
+There are a number of ways to retrieve content from the `TextArea`:
+
+- The [`TextArea.text`][textual.widgets._text_area.TextArea.text] property returns all content in the text area as a string.
+- The [`TextArea.selected_text`][textual.widgets._text_area.TextArea.selected_text] property returns the text corresponding to the current selection.
+- The [`TextArea.get_text_range`][textual.widgets._text_area.TextArea.get_text_range] method returns the text between two locations.
+
+In all cases, when multiple lines of text are retrieved, the [document line separator](#line-separators) will be used.
+
+### Editing content inside `TextArea`
+
+The content of the `TextArea` can be updated using the [`replace`][textual.widgets._text_area.TextArea.replace] method.
+This method is the programmatic equivalent of selecting some text and then pasting.
+
+All atomic (single-cursor) edits can be represented by a `replace` operation, but for
```

Comment:
I wonder if this is getting a little too into the implementation detail of the widget and might mean that the casual reader misses out on the significance of the availability of `insert`, `delete` and `clear`.

Reply:
Agreed, removed that initial part.